File: cpp/ecbenchmark/src/cec10/F19.cpp
#include "ecbenchmark/cec10/F19.h"
#include "ecbenchmark/cec10/CecMath.h"
#include "ecbenchmark/operator/Shifted.h"
#include "ecbenchmark/function/Schwefel1_2.h"


namespace ecb {
    namespace cec10 {

        F19::F19(int dimensions, int mValue)
        : CecFunction("F19", "Shifted Schwefel's Function", dimensions,
        -100, 100, true, mValue, new CecRandom(19l)) {

            std::vector<scalar>  shift = CecMath::ShiftVector(dimensions, minimumDomain(), maximumDomain(), randomizer());
            schwefel = new Shifted(shift, 0, new Schwefel1_2());

        }

        F19::~F19() {
            delete schwefel;
        }

        scalar F19::f(const std::vector<scalar>& x) {
            return schwefel->f(x);
        }
    }
}

File: java/ecbenchmark/src/ecbenchmark/cec10/F19.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package ecbenchmark.cec10;

import ecbenchmark.function.Schwefel;
import ecbenchmark.Function;
import ecbenchmark.operator.Shifted;

/**
 *
 * @author jcrada
 */
public class F19 extends CecFunction {

    private Function schwefel;

    public F19(int dimensions, int mValue) {
        super("F19", "Shifted Schwefel's Function", dimensions,
                -100, 100, true, new CecRandom(19l), mValue);

        double[] shift = CecMath.ShiftVector(dimensions, getMinimum(), getMaximum(), getRandomizer());
        schwefel = new Shifted(shift, 0, new Schwefel());
    }

    public double f(final double[] x) {
        return schwefel.f(x);
    }
}